backend/src/main/java/com/redhat/cloud/notifications/db/repositories/DrawerNotificationRepository.java

```diff
@@ -64,9 +63,8 @@ public List<DrawerEntryPayload> getNotifications(String orgId, String username,
 
     public Long count(String orgId, String username, Set<UUID> bundleIds, Set<UUID> appIds, Set<UUID> eventTypeIds,
                       LocalDateTime startDate, LocalDateTime endDate, Boolean readStatus) {
-
-        String hql = "SELECT count(dn.id) "
-            + "FROM DrawerNotification dn where dn.orgId = :orgId and dn.userId = :userid";
+        String hql = "SELECT COUNT(*) FROM DrawerNotification dn "
+                        + "inner join dn.event as ev where dn.orgId = :orgId and dn.userId = :userid";
```

Comment:
Same question for this `inner join`.

Reply:
We don't need the inner join with `dn.event` here because we're not accessing the event table during the query execution.

Reply:
I'd remove the `inner join` here because it will not be needed in for all queries and I think we can trust Hibernate 6 to make the right decision about the join type.